src/vs/workbench/contrib/chat/browser/chatEditorInput.ts

```diff
@@ -70,9 +71,28 @@ export class ChatEditorInput extends EditorInput implements IEditorCloseHandler
 		this.sessionId = (options.target && 'sessionId' in options.target) ?
 			options.target.sessionId :
 			undefined;
-		this.inputCount = ChatEditorInput.getNextCount();
-		ChatEditorInput.countsInUse.add(this.inputCount);
-		this._register(toDisposable(() => ChatEditorInput.countsInUse.delete(this.inputCount)));
+
+		// Check if we already have a custom title for this session
+		const hasExistingCustomTitle = this.sessionId && (
```

Comment:
This seems wrong, it isn't looking at whether it has a custom title, just any title?